src/operation/emulation_model.jl

```diff
@@ -238,6 +238,27 @@ function build_pre_step!(model::EmulationModel)
             @info "EmulationProblem status not BuildStatus.EMPTY. Resetting"
             reset!(model)
         end
+        # Temporary while are able to switch from PJ to POI
+        container = get_optimization_container(model)
+        container.built_for_recurrent_solves = true
+
+        @info "Initializing Optimization Container For an EmulationModel"
+        init_optimization_container!(
+            get_optimization_container(model),
+            get_network_formulation(get_template(model)),
+            get_system(model),
+        )
+        @info "Initializing ModelStoreParams"
+        init_model_store!(model)
+
+        @info "Mapping Service Models"
+        populate_aggregated_service_model!(get_template(model), get_system(model))
+        populate_contributing_devices!(get_template(model), get_system(model))
+        add_services_to_device_model!(get_template(model))
+
+        @info "Intilization Model"
+        build_initialization!(model)
+        initialize!(model)
```

Comment:
You could probably combine this function with build_initialization.

Reply:
I will merge this with a better name.